pkg/ipam/node.go

```diff
@@ -596,11 +596,84 @@ func (n *Node) maintainIPPool(ctx context.Context) error {
 
 	scopedLog := n.logger()
 
-	// Release excess addresses
-	if a.release != nil && len(a.release.IPsToRelease) > 0 {
+	// Attempt to release if :
+	// * IP is still marked for release in the current iteration's a.release.IPsToRelease
+	// * Marked for release for more than excess-ip-release-delay seconds
+	// * IP is not in use by any pod according to pod informer cache
+	var ipsToRelease []string
+	if n.ipsMarkedForRelease != nil && len(n.ipsMarkedForRelease) > 0 {
+		if a.release == nil || len(a.release.IPsToRelease) == 0 {
+			n.ipsMarkedForRelease = make(map[string]time.Time)
```

Comment:
I'm not sure I understand if clearing `n.ipsMarkedForRelease` is intentional. A comment would help me :)

Reply:
OK, I think I get it: what this is saying is "there were IPs marked for release in a previous iteration, but new maintenance action tells me there are no IPs to release, so `n.ipsMarkedForRelease` is obsolete and can be zapped". Right?

Reply:
And if so, why not move this out of the preceding `if`?